addons/base_import/static/src/js/import_action.js

```diff
@@ -783,79 +977,73 @@ var DataImport = AbstractAction.extend({
-                    var rowName = '';
-                    if (results.name.length > rows.from && results.name[rows.from] !== '') {
-                        rowName = _.str.sprintf(' (%s)', results.name[rows.from]);
-                    }
-                    if (from === to) {
-                        return _.str.sprintf(_t("Row %d%s"), from, rowName);
-                    }
-                    return _.str.sprintf(_t("Between rows %d and %d"),
-                                         from, to);
-                },
-                at_multi_header: function (numberLines) {
-                    return _.str.sprintf(_t("at %d different rows:"),
-                                         numberLines);
-                },
-                more: function (n) {
-                    return _.str.sprintf(_t("(%d more)"), n);
-                },
-                info: function (msg) {
-                    if (typeof msg === 'string') {
-                        return _.str.sprintf(
-                            '<div class="oe_import_moreinfo oe_import_moreinfo_message">%s</div>',
-                            _.str.escapeHTML(msg));
-                    }
-                    if (msg instanceof Array) {
-                        return _.str.sprintf(
-                            '<div class="oe_import_moreinfo oe_import_moreinfo_choices"><a href="#" class="oe_import_report_see_possible_value oe_import_see_all"><i class="fa fa-arrow-right"/> %s </a><ul class="oe_import_report_more">%s</ul></div>',
-                            _.str.escapeHTML(_t("See possible values")),
-                            _(msg).map(function (msg) {
-                                return '<li>'
-                                    + _.str.escapeHTML(msg)
-                                + '</li>';
-                            }).join(''));
-                    }
-                    // Final should be object, action descriptor
-                    return [
-                        '<div class="oe_import_moreinfo oe_import_moreinfo_action">',
-                            _.str.sprintf('<a href="#" data-action="%s" class="oe_import_see_all"><i class="fa fa-arrow-right"/> ',
-                                    _.str.escapeHTML(JSON.stringify(msg))),
-                                _.str.escapeHTML(
-                                    _t("See possible values")),
-                            '</a>',
-                        '</div>'
-                    ].join('');
-                },
-            }));
+
+        // regroup error message by field
+        var messagesGroupedByField = {};
+        _.each(messagesSorted, function (errors) {
+            var field = errors[0].field_path ? errors[0].field_path.join('/') : errors[0].field || 0;
+            [errors] = errors.length > 1 ? [errors] : errors;
+            // regroup errors for same value at different rows in the same error.
+            if (Array.isArray(errors) && errors[0].rows) {
```

Comment:
I have so many questions.
What is rows.from?
What is rows.to?
Why do you play with index so much (errors[0], errors[errors.length - 1], ...)
What is your input structure? And why is it apparently so unnecessarily complicated?
What is love?
(Baby don't hurt me, don't hurt me, no more!).
(Oula, ça va plus).